cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -275,6 +275,10 @@
     /var/lib/snapd/hostfs/var/lib/lxd r,
 
     # support for the mount namespace sharing
+    capability sys_ptrace,
+    # allow snap-confine to read /proc/1/ns/mnt
+    ptrace peer=unconfined,
```

Comment:
From my quick testing using `readlink /proc/1/ns/mnt` and `nsenter --mount=/proc/1/ns/mnt`, you can get away with specifying a single ptrace permission rather than granting all ptrace permissions:

  ```ptrace trace peer=unconfined,```

Also, I don't seem to need `capability sys_ptrace,` but trust that you added it after seeing a denial for that capability.